src/macaron/slsa_analyzer/checks/check_result.py

```diff
@@ -24,25 +29,54 @@ class CheckResultType(str, Enum):
     UNKNOWN = "UNKNOWN"
 
 
-class CheckResult(TypedDict):
-    """This class stores the result of a check in a dictionary."""
+@dataclass
+class CheckInfo:
+    """This class identifies and describes a check."""
 
     check_id: str
     check_description: str
-    # The string representations of the slsa requirements and their
-    # corresponding slsa level.
-    slsa_requirements: list[str]
+    eval_reqs: list[ReqName]
+
+
+@dataclass
+class CheckResultData:
+    """This class stores the result of a check."""
+
     # If an element in the justification is a string,
     # it will be displayed as a string, if it is a mapping,
     # the value will be rendered as a hyperlink in the html report.
-    justification: list[str | dict[str, str]]
+    justification: Justification
     # human_readable_justification: str
     # result_values: dict[str, str | float | int] | list[dict[str, str | float | int]]
-    result_tables: list[DeclarativeBase | Expectation]
+    result_tables: ResultTables
     # recommendation: str
     result_type: CheckResultType
 
 
+@dataclass
+class CheckResult:
+    """This class stores the result of a check, including the description of the check that produced it."""
+
+    check: CheckInfo
+    result: CheckResultData
+
+    def get_dict(self) -> dict:
+        """Get a flattened dictionary representation for this CheckResult.
+
+        Returns
+        -------
+        dict
+        """
+        return {
+            "check_id": self.check.check_id,
+            "check_description": self.check.check_description,
+            "slsa_requirements": [str(BUILD_REQ_DESC.get(req)) for req in self.check.eval_reqs],
+            "justification": self.result.justification,
+            "result_tables": self.result.result_tables,
+            "result_type": self.result.result_type,
+        }
```

Comment:
```suggestion

    def get_dict(self) -> dict:
        """Get a flattened dictionary representation for this CheckResult.
        Returns
        -------
        dict
        """
        return self.check.asdict() | self.result.asdict()
```

Reply:
I don't think `asdict(self.check) | asdict(self.result)` would be equivalent to the original implementation because it don't handle this logic `"slsa_requirements": [str(BUILD_REQ_DESC.get(req)) for req in self.check.eval_reqs],` in the original implementation. Are we meant to remove this from the returned dictionary?

Reply:
Yes, that would change the behaviour. The purpose of this method is to produce the output report format representation, which isn't necessarily the same as an `asdict` of all of the fields. Even if it was the same, being explicit about the output-formatted dict being created makes it easier to change or refactor the representation of this class without accidentally changing the user-visible report format.

Reply:
That could be an indication of unnecessary data being stored in the results and or with an undesirable type. I would avoid duplicating data and adjust the original interface instead, i.e., the attributes in `CheckInfo` and `CheckResultData`.

Reply:
The conversion here is from `eval_reqs` which is a `list[ReqName]` (i.e. list of references to the SLSA requirements, referred to via their identifer enum value) to what the output format expects, which is `slsa_requirements`, containing a list of the SLSA requirements as a textual description. Originally, this conversion happened in the constructor for `CheckResult`, however, I removed it from there because repeatedly storing the SLSA req textual description for each one is redundant since with a ReqName you can look up the description at any time, so there's no need for `CheckResult` to denormalise the data in this way. However, the existing output format has the info in that form (whether this is desirable is another question, the it repeats the same strings but makes the report easier to read, and in any case these changes were intended to be a refactoring and I was not proposing any changes to the reporting format), so the conversion had to happen at some point, so I moved it later to `get_dict` which I considered to be part of the "output layer".

In general, I don't think it is necessarily desirable that the internal representation of results must exactly match the format produced for the reports, having an explicit report generation step that is decoupled from the internal representation makes it easier to have alternative reporting formats (for example, the verification summary could end up being an alternative report format).

Reply:
Agreed that changing the output format can be out of scope for this PR, but I still prefer to clean the data and avoid duplication as much as possible (maybe in future) unless really necessary.

Can you please rename the function from `get_dict()` to `get_summary()` and adjust the docstring to reflect the intent you discuss here?

Reply:
Done.